src/cache.ts

```diff
@@ -53,6 +53,48 @@ const wrapCacheEntry = (entry: CacheEntry): CachePayload => {
   };
 };
 
+/**
+ * Finds the corresponding key in the cache based on the provided key data.
+ * The keys inside the cache are in the format {prefix}::{client_id}::{audience}::{scope}.
+ * The first key in the cache that satisfies the following conditions is returned
+ *  - `prefix` is strict equal to Auth0's internally configured `keyPrefix`
+ *  - `client_id` is strict equal to the `cacheKeyData.client_id`
+ *  - `audience` is strict equal to the `cacheKeyData.audience`
+ *  - `scope` contains at least all the `cacheKeyData.scope` values
+ *  *
+ * @param cacheKeyData The provided cacheKeyData
+ * @param existingCacheKeys A list of existing cache keys
+ */
+const findExistingCacheKey = (
+  cacheKeyData: CacheKeyData,
+  existingCacheKeys: Array<string>
+) => {
+  const { client_id, audience, scope } = cacheKeyData;
+  return existingCacheKeys.find(key => {
```

Comment:
`Array.find` will need a polyfill for IE11 and I'm not sure we _currently_ provide that - can you check?

If not, given our recent work into the state of our polyfills, it would undesirable to add more so we should seek an alternative.

Reply:
I will rework to use filter.

Reply:
Reworked to use `filter()[0]` instead.